database/service_db_test.go

```diff
@@ -60,22 +76,68 @@ func TestServiceDBDelete(t *testing.T) {
 	db, closer := openServiceDB(t)
 	defer closer()
 
-	s := &service.Service{ID: "1", Name: "test-service"}
+	// id
+	s := &service.Service{ID: "00", Alias: "2", Name: "test-service"}
 	require.NoError(t, db.Save(s))
 	require.NoError(t, db.Delete(s.ID))
 	_, err := db.Get(s.ID)
-	require.True(t, IsErrNotFound(err))
+	require.IsType(t, &ErrNotFound{}, err)
+	_, err = db.Get(s.Alias)
+	require.IsType(t, &ErrNotFound{}, err)
+
+	_, err = db.db.Get([]byte(idKeyPrefix+s.ID), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+	_, err = db.db.Get([]byte(aliasKeyPrefix+s.Alias), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+
+	// alias
+	s = &service.Service{ID: "11", Alias: "3", Name: "test-service"}
+	require.NoError(t, db.Save(s))
+	require.NoError(t, db.Delete(s.Alias))
+	_, err = db.Get(s.Alias)
+	require.IsType(t, &ErrNotFound{}, err)
+	_, err = db.Get(s.ID)
+	require.IsType(t, &ErrNotFound{}, err)
+
+	_, err = db.db.Get([]byte(idKeyPrefix+s.ID), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+	_, err = db.db.Get([]byte(aliasKeyPrefix+s.Alias), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+}
+
+func TestServiceDBDeleteConcurrency(t *testing.T) {
+	db, closer := openServiceDB(t)
+	defer closer()
+
+	s := &service.Service{ID: "00", Alias: "2", Name: "test-service"}
+	db.Save(s)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	// service should not be found
+	go func() {
```

Comment:
There is a chance that goroutine might complete before. Instead of checking _NotFound_ error here we should push it to an error slice and do the same for the other call to _Delete_. After _Wait_ done, we should compare the length of slice to see if it's equal to 1 (GOOD) or 2.

Reply:
I understand what you mean but is it really necessary? In this case, it's a go routine versus the main routine. In the case of 2 go routine, then it will make sense. Maybe I should edit the code to call Delete from 2 go routine to have better concurrency?
@krhubert what do you think?

Reply:
Actually, this is still two different goroutines, it doesn't matter that one is main. There are no difference.